editor/editor_node.h

```diff
@@ -319,6 +319,8 @@ class EditorNode : public Node {
 	PopupMenu *editor_layouts;
 	EditorLayoutsDialog *layout_dialog;
 
+	Ref<ConfigFile> window_config;
```

Comment:
Since saving overrides everything regardless of previous values, I think it might not need to be a member variable, you can just instantiate a ConfigFile when loading, and another one when saving.